### ISSUE TITLE
Gradients do not exist for variables ['clustering/clusters:0'] when minimizing the loss.

### ISSUE THREAD

**Issue by zzdj11** [Philyzh8/scTAG#1]
I tried to change to another dataset, but tensorflow came up with the following warning during training and couldn't find a solution. This bothers me a lot.
WARNING:tensorflow:Gradients do not exist for variables ['clustering/clusters:0'] when minimizing the loss.

**Comment by Philyzh8** [comment]
Are you using single-cell data? scTAG is more suitable for single-cell data with strong sparsity. If not, you can also do some feature selection processing on the data.

**Comment by zzdj11** [comment]
> Are you using single-cell data? scTAG is more suitable for single-cell data with strong sparsity. If not, you can also do some feature selection processing on the data.
Thanks for your answer! I am using spatial resolved transcriptomics data, which also has strong sparsity. What do you suggest I should check to resolve this warning?
The warning code appears as the gradient update statement: optimizer.apply_gradients(zip(grads, vars))

**Comment by tb1over** [comment]
i think you should compare your model performance with scDeepCluster,  which you cited.  your work add a graph autoencoder module baes the scDeepCluster
